src/lib.rs

```diff
@@ -4562,13 +4579,27 @@ macro_rules! transmute_ref {
 /// Safely transmutes a mutable reference of one type to a mutable reference of
 /// another type of the same size.
 ///
-/// The expression `$e` must have a concrete type, `&mut T`, where `T: Sized +
-/// IntoBytes`. The `transmute_mut!` expression must also have a concrete type,
-/// `&mut U` (`U` is inferred from the calling context), where `U: Sized +
-/// FromBytes`. It must be the case that `align_of::<T>() >= align_of::<U>()`.
+/// This macro behaves like an invocation of this function:
 ///
-/// The lifetime of the input type, `&mut T`, must be the same as or outlive the
-/// lifetime of the output type, `&mut U`.
+/// ```ignore
+/// const fn transmute_mut<'src, 'dst, Src, Dst>(src: &'src mut Src) -> &'dst mut Dst
+/// where
+///     'src: 'dst,
+///     Src: IntoBytes + FromBytes + NoCell,
+///     Dst: FromBytes + IntoBytes + NoCell,
```

Comment:
```suggestion
///     Src: FromBytes + IntoBytes + NoCell,
///     Dst: FromBytes + IntoBytes + NoCell,
```

IMO it's visually clearer this way that both types must implement the same set of traits.

Reply:
Done